### DISCUSSION TITLE
Have we done this improvement?

### DISCUSSION THREAD

**Discussion by wangzhen11aaa** [databendlabs/databend#4072] [Q&A] [answered]
FYR.

https://github.com/ClickHouse/ClickHouse/pull/28272

**Comment by Xuanwo** [comment]
ping @sundy-li, can you take a look at the question?

**Comment by sundy-li** [comment, marked as answer]
Yes, it's already done I think.  Just combine the bitmaps with filters.

In  https://github.com/datafuselabs/databend/blob/main/common/functions/src/aggregates/aggregate_combinator_if.rs#L111-L118

**Reply by wangzhen11aaa** [reply]
c~
